lib/owncloud/test_sharePropagationGroups.py

```diff
@@ -0,0 +1,320 @@
+| 11          | verify etag is the same | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 12          |                         | R2 upload in shared dir |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 13          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 14          |                         |                         | upload in shared dir |
++-------------+-------------------------+-------------------------+----------------------+
+| 15          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 16          |                         | R2 unshares folder      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 17          | verify etag is the same | verify etag is the same | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+"""
+
+from smashbox.utilities import *
+import itertools
+import os.path
+import re
+import operator as op
+
+def get_group_name(i):
+    return '%s%i' % (config.oc_group_name, i)
+
+def get_account_name(i):
+    return '%s%i' % (config.oc_account_name, i)
+
+group_map = {
+    # maps the group name with the usernum belonging to the group
+    get_group_name(1) : [2,3],
+    get_group_name(2) : [4,5],
+    get_group_name(3) : [6,7],
+}
+
+def compare_list(list1, list2, func):
+    """
+    Compare the list item by item using the function func. If func returns False, compare list
+    will return False
+    """
+    if len(list1) != len(list2):
+        return False
+
+    for index in range(0, len(list1)):
+        if not func(list1[index], list2[index]):
+            return False
+    return True
+
+def get_client_etags(clients):
+    new_etags = []
+    for client in clients:
+        new_etags.append(client.file_info('/').get_etag())
+
+    return new_etags
+
+def run_group_ocsync(d, group_name):
+    for usernum in group_map[group_name]:
+        run_ocsync(os.path.join(d, str(usernum)), user_num=usernum)
+
+def parse_worker_number(worker_name):
+    match = re.search(r'(\d+)$', worker_name)
+    if match is not None:
+        return int(match.group())
+    else:
+        return None
+
+@add_worker
+def setup(step):
+
+    step(1, 'create test users')
+    reset_owncloud_account(num_test_users=7)
+    reset_owncloud_group(num_groups=3)
+    check_users(7)
+
+    for group in group_map:
+        for user in group_map[group]:
+            add_user_to_group(get_account_name(user), group)
+
+    reset_rundir()
```

Comment:
please remove everthing from here `reset_rundir()` down to the end of the worker, smashbox takes care of that:
https://github.com/owncloud/smashbox/commit/79006cbc977c3232d3ea117055c9dd2f7fabdcb8